task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -685,7 +685,9 @@ def _service_subprocess(
         :param expect_signal: Signal not to log if the task exits with this code.
         :returns: The process exit code, or None if it's still alive
         """
-        events = self.selector.select(timeout=max_wait_time)
+        # Ensure minimum timeout to prevent CPU spike with tight loop when timeout is 0 or negative
+        timeout = max(0.01, max_wait_time)
+        events = self.selector.select(timeout=timeout)
```

Comment:
This change could have been in `_monitor_subprocess(` but `_service_subprocess` is called from 2 places: 1) `_monitor_subprocess(` and `kill` ( [code](https://github.com/apache/airflow/blob/65fac1a9fa5fee59ff88aaddcd7a6154bd348dd8/task-sdk/src/airflow/sdk/execution_time/supervisor.py#L596) ) -- so I did it here

https://github.com/apache/airflow/blob/65fac1a9fa5fee59ff88aaddcd7a6154bd348dd8/task-sdk/src/airflow/sdk/execution_time/supervisor.py#L876C9-L895